appIcons.js

```diff
@@ -1225,77 +1225,65 @@ function getInterestingWindows(windows, monitorIndex) {
 }
 
 /**
- * A ShowAppsIcon improved class.
- *
- * - set label position based on dash orientation (Note, I am reusing most machinery of the appIcon class)
- * - implement a popupMenu based on the AppIcon code (Note, I am reusing most machinery of the appIcon class)
- *
+ * Use AppIcon to handle the menu for the dock icon
  */
 
 var DockShowAppsIcon = GObject.registerClass({
     Signals: {
         'menu-state-changed': { param_types: [GObject.TYPE_BOOLEAN] },
-        'sync-tooltip': {}
-    }
-}
-, class DockShowAppsIcon extends Dash.ShowAppsIcon {
-    _init() {
-        super._init();
-
-        // Re-use appIcon methods
-        let appIconPrototype = AppDisplay.AppIcon.prototype;
-        this.toggleButton.y_expand = false;
-        this.toggleButton.connect('popup-menu',
-            appIconPrototype._onKeyboardPopupMenu.bind(this));
-        this.toggleButton.connect('clicked',
-            this._removeMenuTimeout.bind(this));
-
-        this.reactive = true;
-        this.toggleButton.popupMenu = () => this.popupMenu.call(this);
-        this.toggleButton._removeMenuTimeout = () => this._removeMenuTimeout.call(this);
+        'sync-tooltip': {},
+    },
+}, class DockShowAppsIcon extends AppDisplay.AppIcon {
+    _init(source) {
+        // Skip AppIcon._init to avoid the app handling logic
+        AppDisplay.AppViewItem.prototype._init.call(this, {}, false, false);
+
+        // Keep a reference to the container
+        this.source = source;
+
+        this._iconActor = null;
+        
+        // Create the icon
+        this.icon = new IconGrid.BaseIcon(_('Show Applications'), {
+            setSizeManually: true,
+            showLabel: false,
+            createIcon: this._createIcon.bind(this),
+        });
+        this.icon.y_align = Clutter.ActorAlign.CENTER;
```

Comment:
I have my dock in **left** panel mode, and I believe this `CENTER` change is the reason my icon is no longer at the end (bottom). Could that orientation be detected here?

(I'm here because Fedora applied your patch to its package.)